### ISSUE TITLE
Suggest "Prefer AsSpan over Substring" (CA1846) for equals comparisons and other methods

### ISSUE THREAD

**Issue by xPaw** [dotnet/roslyn-analyzers#7362]
### Analyzer

**Diagnostic ID**: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1846

### Describe the improvement

When comparing a substring with equals sign, or calling methods that have an equivalent overload for ReadOnlySpan<char> I would like this rule to suggest using spans.
```
        public bool Foo(string str)
        {
            if (str[1..] == "test") { }
            if (str.Substring(1) == "test") { }
            if (str.Substring(1).Equals("test", StringComparison.Ordinal)) { }
            if (str.Substring(1).StartsWith("test")) { }
            Consume(str.Substring(1));
        }

        public void Consume(string str) { }
        public void Consume(ReadOnlySpan<char> str) { }
```

In this example only `Consume` triggers this warning.

**Comment by buyaa-n** [comment]
Thank you for reporting the issue @xPaw, I am supportive with the enhancement, some scenarios might need a new analyzer instead of updating [ca1846](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1846), anyway it would be helpful for contributors to see the proposed fixes for all cases.
